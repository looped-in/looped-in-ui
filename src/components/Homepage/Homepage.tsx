import Footer from "@/components/Footer";
import Header from "@/components/Header";
import TopBannerHomepage from "@/components/TopBannerHomepage";
import React from "react";
import FindBuilder from "../FindBuilder";
import Processes from "../Processes";

const Homepage = () => {
	return (
		<div>
			<Header />

			<TopBannerHomepage />

			<FindBuilder />

			<Processes />

			<Footer />
		</div>
	);
};

export default Homepage;
