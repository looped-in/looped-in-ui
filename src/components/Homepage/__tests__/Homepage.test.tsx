import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import Homepage from "@/components/Homepage"

describe("Homepage", () => {
  it("SHOULD match snapshot", () => {
    const { container } = render(<Homepage />)
    expect(container).toMatchSnapshot()
  })

  it("SHOULD renders correct heading", () => {
    render(<Homepage />)
    expect(
      screen.getByText(
        "Collaboration tools you'll love for the jobs you don't."
      )
    ).toBeInTheDocument()
    expect(
      screen.getByText("Find a builder for your next job!")
    ).toBeInTheDocument()
    expect(screen.getByText("Loopedin the right way!")).toBeInTheDocument()
  })
})
