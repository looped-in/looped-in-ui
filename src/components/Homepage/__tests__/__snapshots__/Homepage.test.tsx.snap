// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage SHOULD match snapshot 1`] = `
<div>
  <div
    class="bg-white"
  >
    <div
      class=" relative flex flex-row px-28 pb-28 space-x-4"
    >
      <div
        class="bg-gradient-to-tl from-secondary to-primary absolute bottom-0 right-0 w-1/2 h-full rounded-full opacity-30 blur-3xl"
      />
      <div
        class="w-1/2 space-y-6 flex flex-col justify-center z-10"
      >
        <div
          class="text-4xl font-medium"
        >
          Collaboration tools you'll love for the jobs you don't.
        </div>
        <div
          class="text-sm"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Luctus morbi est porttitor velit, sit bibendum vitae blandit. Viverra orci ornare vehicula mauris purus pulvinar diam, vel. Odio pharetra mauris metus scelerisque.
        </div>
        <div
          class="text-sm flex flex-row justify-between"
        >
          <a
            class="w-1/3"
            href="/sign-in"
          >
            <button
              class="h-10 w-32 rounded-tl-lg rounded-tr-lg rounded-br-lg bg-gradient-to-b from-secondary to-primary text-white font-bold"
            >
              Sign in
            </button>
          </a>
          <div
            class="text-center w-1/2 content-around font-bold"
          >
            <a
              class="flex flex-row align-bottom gap-2"
              href="/find-builder"
            >
              <svg
                class="lucide lucide-search"
                fill="none"
                height="16"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <path
                  d="m21 21-4.3-4.3"
                />
              </svg>
              <div>
                Find a Builder
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="w-1/2 flex flex-col gap-7 z-10"
      >
        <div
          class=""
        >
          <img
            alt="milestones"
            class="rounded-xl"
            data-nimg="1"
            decoding="async"
            height="273"
            loading="lazy"
            src="/_next/image?url=%2Ftop-banner-milestones.png&w=750&q=75"
            srcset="/_next/image?url=%2Ftop-banner-milestones.png&w=384&q=75 1x, /_next/image?url=%2Ftop-banner-milestones.png&w=750&q=75 2x"
            style="color: transparent;"
            width="359"
          />
        </div>
        <div
          class="flex-row flex justify-start gap-7 "
        >
          <div
            class="w-1/2"
          >
            <img
              alt="updates"
              class="rounded-xl"
              data-nimg="1"
              decoding="async"
              height="286"
              loading="lazy"
              src="/_next/image?url=%2Ftop-banner-updates.png&w=640&q=75"
              srcset="/_next/image?url=%2Ftop-banner-updates.png&w=384&q=75 1x, /_next/image?url=%2Ftop-banner-updates.png&w=640&q=75 2x"
              style="color: transparent;"
              width="274"
            />
          </div>
          <div
            class="w-1/2"
          >
            <img
              alt="financials"
              class="rounded-xl"
              data-nimg="1"
              decoding="async"
              height="228"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Ftop-banner-financials.png&w=3840&q=75"
              srcset="/_next/image?url=%2Ftop-banner-financials.png&w=640&q=75 640w, /_next/image?url=%2Ftop-banner-financials.png&w=750&q=75 750w, /_next/image?url=%2Ftop-banner-financials.png&w=828&q=75 828w, /_next/image?url=%2Ftop-banner-financials.png&w=1080&q=75 1080w, /_next/image?url=%2Ftop-banner-financials.png&w=1200&q=75 1200w, /_next/image?url=%2Ftop-banner-financials.png&w=1920&q=75 1920w, /_next/image?url=%2Ftop-banner-financials.png&w=2048&q=75 2048w, /_next/image?url=%2Ftop-banner-financials.png&w=3840&q=75 3840w"
              style="color: transparent; width: 100%; height: auto;"
              width="300"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class=" flex flex-row h-3/6"
    >
      <div
        class="w-3/6"
      >
        <img
          alt="find-builder"
          data-nimg="1"
          decoding="async"
          height="1500"
          loading="lazy"
          src="/_next/image?url=%2Ffind-builder.png&w=3840&q=75"
          srcset="/_next/image?url=%2Ffind-builder.png&w=1920&q=75 1x, /_next/image?url=%2Ffind-builder.png&w=3840&q=75 2x"
          style="color: transparent;"
          width="1500"
        />
      </div>
      <div
        class="w-3/6 bg-zinc-800 content-around p-8 space-y-6"
      >
        <div
          class="text-transparent bg-clip-text bg-gradient-to-b from-secondary to-primary text-4xl font-semibold"
        >
          Find a builder for 
          <br />
          your next job!
        </div>
        <div
          class="text-slate-50 text-sm "
        >
          Building or renovating? Let Loopedin help you find the right builder for your next big project. Browse to find your builder!
        </div>
        <div>
          <a
            href="/find-builder"
          >
            <button
              class="h-10 rounded-tl-lg rounded-tr-lg rounded-br-lg bg-gradient-to-b from-secondary to-primary text-white font-bold w-1/2"
            >
              Find a Builder
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col py-20"
    >
      <div
        class="text-center text-6xl font-semibold"
      >
        Loopedin the right way!
      </div>
      <div
        class="flex flex-row py-24 pl-28 gap-40"
      >
        <div
          class="w-1/2 space-y-6 p-4 place-content-center"
        >
          <button
            class="h-10 w-32 rounded-tl-lg rounded-tr-lg rounded-br-lg bg-gradient-to-b from-secondary to-primary text-white font-bold"
          >
            Step 1
          </button>
          <div
            class="text-2xl font-semibold"
          >
            Step One of The Process
          </div>
          <div
            class="text-sm"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Luctus morbi est porttitor velit, sit bibendum vitae blandit. Viverra orci ornare vehicula mauris purus pulvinar diam, vel. Odio pharetra mauris metus scelerisque. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Luctus morbi est porttitor velit, sit bibendum vitae blandit. Viverra orci ornare vehicula mauris purus pulvinar diam, vel. Odio pharetra mauris metus scelerisque.
          </div>
        </div>
        <div
          class="w-1/2 relative"
        >
          <div
            class="bg-gradient-radial from-secondary to-primary absolute w-full h-full rounded-full opacity-20 blur-3xl z-0"
          />
          <div
            class="flex"
          >
            <img
              alt="step-1"
              class="z-10"
              data-nimg="1"
              decoding="async"
              height="1000"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fstep-1.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fstep-1.png&w=640&q=75 640w, /_next/image?url=%2Fstep-1.png&w=750&q=75 750w, /_next/image?url=%2Fstep-1.png&w=828&q=75 828w, /_next/image?url=%2Fstep-1.png&w=1080&q=75 1080w, /_next/image?url=%2Fstep-1.png&w=1200&q=75 1200w, /_next/image?url=%2Fstep-1.png&w=1920&q=75 1920w, /_next/image?url=%2Fstep-1.png&w=2048&q=75 2048w, /_next/image?url=%2Fstep-1.png&w=3840&q=75 3840w"
              style="color: transparent; width: 100%; height: auto;"
              width="1000"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-row py-24 pl-28 gap-40"
      >
        <div
          class="w-1/2 relative"
        >
          <div
            class="bg-gradient-radial from-secondary to-primary absolute w-2/3 h-2/3 right-0 bottom-0 rounded-full opacity-20 blur-3xl -z-0"
          />
          <div
            class="flex flex-row-reverse"
          >
            <img
              alt="step-2"
              class="z-10"
              data-nimg="1"
              decoding="async"
              height="500"
              loading="lazy"
              src="/_next/image?url=%2Fstep-2.1.png&w=828&q=75"
              srcset="/_next/image?url=%2Fstep-2.1.png&w=640&q=75 1x, /_next/image?url=%2Fstep-2.1.png&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
          <div
            class="absolute -bottom-20 right-40 z-10"
          >
            <img
              alt="step-2-milestones.png"
              class="z-10"
              data-nimg="1"
              decoding="async"
              height="500"
              loading="lazy"
              src="/_next/image?url=%2Ftop-banner-milestones.png&w=828&q=75"
              srcset="/_next/image?url=%2Ftop-banner-milestones.png&w=640&q=75 1x, /_next/image?url=%2Ftop-banner-milestones.png&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
        </div>
        <div
          class="w-1/2 space-y-6 p-4 place-content-center"
        >
          <button
            class="h-10 w-32 rounded-tl-lg rounded-tr-lg rounded-br-lg bg-gradient-to-b from-secondary to-primary text-white font-bold"
          >
            Step 2
          </button>
          <div
            class="text-2xl font-semibold"
          >
            Step Two of The Process
          </div>
          <div
            class="text-sm"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Luctus morbi est porttitor velit, sit bibendum vitae blandit. Viverra orci ornare vehicula mauris purus pulvinar diam, vel. Odio pharetra mauris metus scelerisque. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Luctus morbi est porttitor velit, sit bibendum vitae blandit. Viverra orci ornare vehicula mauris purus pulvinar diam, vel. Odio pharetra mauris metus scelerisque.
          </div>
        </div>
      </div>
      <div
        class="flex flex-row py-24 pl-28 gap-40"
      >
        <div
          class="w-1/2 space-y-6 p-8 place-content-center"
        >
          <button
            class="h-10 w-32 rounded-tl-lg rounded-tr-lg rounded-br-lg bg-gradient-to-b from-secondary to-primary text-white font-bold"
          >
            Step 1
          </button>
          <div
            class="text-2xl font-semibold"
          >
            Step Three of The Process
          </div>
          <div
            class="text-sm"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Luctus morbi est porttitor velit, sit bibendum vitae blandit. Viverra orci ornare vehicula mauris purus pulvinar diam, vel. Odio pharetra mauris metus scelerisque. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Luctus morbi est porttitor velit, sit bibendum vitae blandit. Viverra orci ornare vehicula mauris purus pulvinar diam, vel. Odio pharetra mauris metus scelerisque.
          </div>
        </div>
        <div
          class="w-1/2 relative"
        >
          <div
            class="bg-gradient-radial from-secondary to-primary absolute w-full h-full right-0 bottom-0 rounded-full opacity-20 blur-3xl"
          />
          <div
            class="flex"
          >
            <div
              class="z-10"
            >
              <img
                alt="step-3"
                data-nimg="1"
                decoding="async"
                height="597"
                loading="lazy"
                src="/_next/image?url=%2Fstep-3.1.png&w=640&q=75"
                srcset="/_next/image?url=%2Fstep-3.1.png&w=384&q=75 1x, /_next/image?url=%2Fstep-3.1.png&w=640&q=75 2x"
                style="color: transparent;"
                width="317"
              />
            </div>
            <div
              class="absolute left-40 top-60 z-10"
            >
              <img
                alt="step-3"
                data-nimg="1"
                decoding="async"
                height="597"
                loading="lazy"
                src="/_next/image?url=%2Ftop-banner-financials.png&w=640&q=75"
                srcset="/_next/image?url=%2Ftop-banner-financials.png&w=384&q=75 1x, /_next/image?url=%2Ftop-banner-financials.png&w=640&q=75 2x"
                style="color: transparent;"
                width="317"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
